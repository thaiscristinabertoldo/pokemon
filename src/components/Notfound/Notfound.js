import React, { memo } from 'react'

import { Title } from '../Title/Title'
import { Image } from '../Image/Image'

import emptylist from '../../assets/emptylist.svg'

import NotfoundStyled from './notfound.css'

export const Notfound = memo(({ message = 'Nenhum registro encontrado' }) => {
  return (
    <NotfoundStyled>
      <Image src={emptylist} name="Nenhum registro encontrado" size="300px" />
      <Title title={message} />
    </NotfoundStyled>
  )
})
