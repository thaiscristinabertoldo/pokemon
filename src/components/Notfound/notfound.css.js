import styled from 'styled-components'

const NotfoundStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  margin-top: 30px;
`

export default NotfoundStyled
