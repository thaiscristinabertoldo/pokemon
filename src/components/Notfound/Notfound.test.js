import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Notfound } from './Notfound'

afterEach(cleanup)

describe('Notfound', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Notfound />)

    expect(asFragment(<Notfound />)).toMatchSnapshot()
  })

  it('should render a src', () => {
    const { getByTestId } = render(<Notfound />)
    expect(getByTestId('image')).toHaveAttribute('src')
  })

  it('should take a message text', () => {
    const { getByText } = render(<Notfound message="Message text" />)

    expect(getByText('Message text')).toBeTruthy()
  })
})
