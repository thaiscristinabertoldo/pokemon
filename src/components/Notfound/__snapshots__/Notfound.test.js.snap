// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notfound should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ dTSlMl"
  >
    <img
      alt="Nenhum registro encontrado"
      data-testid="image"
      height="300px"
      src="emptylist.svg"
      title="Nenhum registro encontrado"
      width="300px"
    />
    <h1
      class="sc-AxjAm byhsde"
    >
      Nenhum registro encontrado
    </h1>
  </div>
</DocumentFragment>
`;
